src/openforms/js/components/admin/form_design/registrations/json_dump/JSONDumpVariableConfigurationEditor.js

```diff
@@ -78,6 +88,30 @@ const JSONDumpVariableConfigurationEditor = ({variable}) => {
           />
         </Field>
       </FormRow>
+
+      {componentType in CUSTOM_COMPONENTS_SCHEMA && (
+        <FormRow>
+          <Field name={`transformToList.${variable.key}`}>
+            <Checkbox
+              name="transformToListCheckbox"
+              label={
+                <FormattedMessage
+                  defaultMessage="Transform to list"
+                  description="'Transform to list' checkbox label"
+                />
+              }
+              helpText={
+                <FormattedMessage
+                  description="'Transform to list' checkbox help text"
+                  defaultMessage="Whether to transform the data of the component to a list or not (depends on the component type)"
+                />
+              }
+              checked={transformToList?.[variable.key] || false}
+              onChange={e => setFieldValue(`transformToList.${variable.key}`, e.target.checked)}
+            />
+          </Field>
```

Comment:
I'd rather just override the `onChange` and do similar stuff like on line 78 instead of working with a mapping, as mentioned above.